File: src/First-App.Core/Entities/Card.cs
using First_App.Core.Abstractions;

namespace First_App.Core.Entities;

public class Card : BaseEntity,
                    ISoftDeleted
{
    public string Name { get; set; } = default!;

    public string Description { get; set; } = default!;

    public DateOnly DueDate { get; set; }

    public bool IsDeleted { get; set; } = false;

    public int? GroupId { get; set; }

    public int PriorityId { get; set; }

    public GroupList? Group { get; set; }

    public Priority? Priority { get; set; }

    public List<Change> ChangeHistory { get; set; } = [];
}

File: src/First-App.WebUI/src/app/common/models/card.ts
export interface Card {
  id: number,
  name: string,
  description: string,
  dueDate: Date,
  priorityId: number,
  groupId: number
}

export function compareCards(a: Card, b: Card) {
  if (a.dueDate > b.dueDate) {
    return 1;
  }
  else if (a.dueDate < b.dueDate) {
    return -1;
  }
  else if (a.id > b.id) {
    return 1;
  }
  else if (a.id < b.id) {
    return -1;
  }
  else {
    return 0;
  }
}
